src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsassessforearlyreleaseapi/resource/AddressResource.kt

```diff
@@ -258,6 +259,53 @@ class AddressResource(private val addressService: AddressService) {
     @Valid @RequestBody addCasCheckRequest: AddCasCheckRequest,
   ) = addressService.addCasCheckRequest(prisonNumber, addCasCheckRequest)
 
+  @DeleteMapping("/offender/{prisonNumber}/current-assessment/address-request/{requestId}")
+  @PreAuthorize("hasAnyRole('ASSESS_FOR_EARLY_RELEASE_ADMIN')")
+  @ResponseStatus(code = HttpStatus.NO_CONTENT)
+  @Operation(
+    summary = "Deletes an address check request for an assessment.",
+    description = "Deletes an address check request for an offender's current assessment.",
+    security = [SecurityRequirement(name = "assess-for-early-release-admin-role")],
+  )
+  @ApiResponses(
+    value = [
+      ApiResponse(
+        responseCode = "204",
+        description = "The address check request has been deleted.",
+        content = [
+          Content(
+            mediaType = "application/json",
+            schema = Schema(implementation = Void::class),
+          ),
+        ],
+      ),
+      ApiResponse(
+        responseCode = "401",
+        description = "Unauthorised, requires a valid Oauth2 token",
+        content = [
+          Content(
+            mediaType = "application/json",
+            schema = Schema(implementation = ErrorResponse::class),
+          ),
+        ],
+      ),
+      ApiResponse(
+        responseCode = "403",
+        description = "Forbidden, requires an appropriate role",
+        content = [
+          Content(
+            mediaType = "application/json",
+            schema = Schema(implementation = ErrorResponse::class),
+          ),
+        ],
+      ),
+    ],
```

Comment:
**question:** do we need a 404 here too?

Reply:
yes put in correct place now :)